From c98a006be077c944612a69545463f1d8a66e8b92 Mon Sep 17 00:00:00 2001
From: Viktor Ashirov <vashirov@redhat.com>
Date: Thu, 11 Dec 2025 12:02:33 +0100
Subject: [PATCH] ipa-pwd-extop: fix bind DN memory leaks in pre-op handlers

In `ipapwd_pre_add()` and `ipapwd_pre_mod()`, `binddn` is obtained via
`slapi_pblock_get()` with SLAPI_CONN_DN which returns an allocated
string. This string was never freed after use.

Add `slapi_ch_free_string(&binddn)` calls after the bind DN is no longer
needed in both functions.

Related: https://pagure.io/freeipa/issue/9895
Signed-off-by: Viktor Ashirov <vashirov@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>
---
 daemons/ipa-slapi-plugins/ipa-pwd-extop/prepost.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/daemons/ipa-slapi-plugins/ipa-pwd-extop/prepost.c b/daemons/ipa-slapi-plugins/ipa-pwd-extop/prepost.c
index b0162c49cd4db9684085c62ada4c30daf657b2e2..8f8d0a0929105869d9df15339d8c447d890f92bf 100644
--- a/daemons/ipa-slapi-plugins/ipa-pwd-extop/prepost.c
+++ b/daemons/ipa-slapi-plugins/ipa-pwd-extop/prepost.c
@@ -379,6 +379,7 @@ static int ipapwd_pre_add(Slapi_PBlock *pb)
                 break;
             }
         }
+        slapi_ch_free_string(&binddn);
     }
 
     pwdop->pwdata.dn = slapi_ch_strdup(slapi_sdn_get_dn(sdn));
@@ -880,6 +881,7 @@ static int ipapwd_pre_mod(Slapi_PBlock *pb)
 
         slapi_sdn_free(&bdn);
         slapi_sdn_free(&tdn);
+        slapi_ch_free_string(&binddn);
     }
 
     pwdop->pwdata.dn = slapi_ch_strdup(slapi_sdn_get_dn(sdn));
-- 
2.52.0

